mlflow/utils/autologging_utils/safety.py

```diff
@@ -360,19 +360,25 @@ def safe_patch_function(*args, **kwargs):
 
             original = gorilla.get_original_attribute(destination, function_name)
 
-            # If the autologging integration associated with this patch is disabled,
-            # call the original function and return
-            if autologging_is_disabled(autologging_integration):
-                return original(*args, **kwargs)
-
-            # Whether or not to exclude auto-autologged content from content explicitly logged via
-            # `mlflow.start_run()`
+            # Whether or not to exclude autologged content from user-created fluent runs
+            # (i.e. runs created manually via `mlflow.start_run()`)
             exclusive = get_autologging_config(autologging_integration, "exclusive", False)
+            user_created_fluent_run_is_active = (
+                mlflow.active_run() and not _AutologgingSessionManager.active_session()
+            )
 
-            active_run = mlflow.active_run()
-
-            if active_run and exclusive and not _AutologgingSessionManager.active_session():
-                return original(*args, **kwargs)
+            if autologging_is_disabled(autologging_integration) or (
+                user_created_fluent_run_is_active and exclusive
+            ):
+                # If the autologging integration associated with this patch is disabled,
+                # or if the current autologging integration is in exclusive mode and a user-created
+                # fluent run is active, call the original function and return. Restore the original
+                # warning behavior during original function execution, since autologging is being
+                # skipped
+                with set_non_mlflow_warnings_behavior_for_current_thread(
```

Comment:
Another option would be to move the context manager entry statements from above down below this logic. However, that leaves a small risk of us leaking logger / warning statements from this section of the code.